components/ide-service/pkg/server/server.go

```diff
@@ -356,20 +359,17 @@ func (s *IDEServiceServer) ResolveWorkspaceConfig(ctx context.Context, req *api.
 	// make a copy for ref ideConfig, it's safe because we replace ref in update config
 	ideConfig := s.code1_85IdeConfig
 
-	var defaultIde *config.IDEOption
-
 	if ide, ok := ideConfig.IdeOptions.Options[ideConfig.IdeOptions.DefaultIde]; !ok {
 		// I think it never happen, we have a check to make sure all DefaultIDE should be in Options
 		log.WithError(err).WithField("defaultIDE", ideConfig.IdeOptions.DefaultIde).Error("IDE configuration corrupt, cannot found defaultIDE")
 		return nil, fmt.Errorf("IDE configuration corrupt")
 	} else {
-		defaultIde = &ide
-	}
-
-	resp = &api.ResolveWorkspaceConfigResponse{
-		SupervisorImage: ideConfig.SupervisorImage,
-		WebImage:        defaultIde.Image,
-		IdeImageLayers:  defaultIde.ImageLayers,
+		// we always need WebImage for when the user chooses a desktop ide
+		resp = &api.ResolveWorkspaceConfigResponse{
```

Comment:
How about make `ide, ideOk := ...` a new line? Put this general logic in else looks weired